app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -344,4 +356,42 @@ def stale_cache?(last_refresh_time)
       last_refresh_time > current_val[:created_at]
     end
   end
+
+  private
+
+  #
+  # Checks if IP address to host name resolving is enabled.
+  #
+  # @return [Boolean] `true` if host name resolving is enabled in the configuration, `false` otherwise.
+  #
+  def resolve_ip_addresses?
+    ::Settings.ems.ems_redhat.resolve_ip_addresses
+  end
+
+  #
+  # Tries to convert the given IP address into a host name, doing a reverse DNS lookup if needed. If it
+  # isn't possible to find the host name the original IP address will be returned, and a warning will be
+  # written to the log.
+  #
+  # @param address [String] The IP address.
+  # @return [String] The host name.
+  #
+  def resolve_ip_address(address)
+    require 'resolv'
```

Comment:
* You do need a require under normal circumstances as while it's part of stdlib, it isn't loaded by default.
* The module is present in our current Rails env, it's not clear as to what actually adds it, probably Rails for it's own purposes. So...
* It's appropriate to require it in this file as it's a required dependency, but I'd put it at the top of the file instead of within the method itself.